rcl/src/rcl/publisher.c

```diff
@@ -124,46 +124,47 @@ rcl_publisher_get_default_options()
 rcl_ret_t
 rcl_publish(const rcl_publisher_t * publisher, const void * ros_message)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(publisher, RCL_RET_INVALID_ARGUMENT);
-  RCL_CHECK_ARGUMENT_FOR_NULL(ros_message, RCL_RET_INVALID_ARGUMENT);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl, "publisher is invalid", return RCL_RET_PUBLISHER_INVALID);
-  if (rmw_publish(publisher->impl->rmw_handle, ros_message) != RMW_RET_OK) {
+  if (!rcl_publisher_is_valid(publisher)) return RCL_RET_PUBLISHER_INVALID;
+  else if (rmw_publish(publisher->impl->rmw_handle, ros_message) != RMW_RET_OK) {
     RCL_SET_ERROR_MSG(rmw_get_error_string_safe());
     return RCL_RET_ERROR;
-  }
-  return RCL_RET_OK;
+  } return RCL_RET_OK;
 }
 
 const char *
 rcl_publisher_get_topic_name(const rcl_publisher_t * publisher)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(publisher, NULL);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl, "publisher is invalid", return NULL);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl->rmw_handle, "publisher is invalid", return NULL);
+  if (!rcl_publisher_is_valid(publisher)) return NULL;
   return publisher->impl->rmw_handle->topic_name;
 }
 
 const rcl_publisher_options_t *
 rcl_publisher_get_options(const rcl_publisher_t * publisher)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(publisher, NULL);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl, "publisher is invalid", return NULL);
+  if (!rcl_publisher_is_valid(publisher)) return NULL;
   return &publisher->impl->options;
 }
 
 rmw_publisher_t *
 rcl_publisher_get_rmw_handle(const rcl_publisher_t * publisher)
 {
-  RCL_CHECK_ARGUMENT_FOR_NULL(publisher, NULL);
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    publisher->impl, "publisher is invalid", return NULL);
+  if (!rcl_publisher_is_valid(publisher)) return NULL;
   return publisher->impl->rmw_handle;
 }
 
+bool
+rcl_publisher_is_valid(const rcl_publisher_t * publisher)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(publisher, false);
+  RCL_CHECK_FOR_NULL_WITH_MSG(publisher->impl,
+                              "publisher implementation is invalid",
+                              return NULL);
```

Comment:
Please wrap all or none of the arguments. I'm not sure if the linter will automatically correct this. This is what I would do:

```c++
RCL_CHECK_FOR_NULL_WITH_MSG(
  publisher->impl, "publisher implementation is invalid", return NULL);
```

Or this:

```c++
RCL_CHECK_FOR_NULL_WITH_MSG(
  publisher->impl,
  "publisher implementation is invalid",
  return NULL);
```